### PR TITLE
Fix decoding of WETH in uniswap v3

### PR DESCRIPTION
**Pull request by yabirgb**
As it was before if a user deposited WETH instead of ETH the decoder was not detecting correctly the deposit event

### CODE REVIEW

File: rotkehlchen/chain/ethereum/modules/uniswap/v3/decoder.py [hunk 1]
```diff
@@ -415,7 +423,6 @@ def _maybe_decode_v3_deposit_or_withdrawal(
                 evm_inquirer=self.evm_inquirer,
                 encounter=TokenEncounterInfo(tx_hash=context.transaction.tx_hash),
             )
-            token_with_data = self.eth if token_with_data == A_WETH else token_with_data
```
Comment (yabirgb) [inline rotkehlchen/chain/ethereum/modules/uniswap/v3/decoder.py:418]: this line was making that the check was always against ETH instead of ETH/WETH. The uniswap pools always have WETH afaik and allow to withdraw/deposit ETH

File: rotkehlchen/chain/ethereum/modules/uniswap/v3/decoder.py [hunk 2]
```diff
@@ -89,6 +89,14 @@ def _find_from_asset_and_amount(events: list['EvmEvent']) -> tuple[Asset, FVal]
     return from_asset, from_amount
 
 
+def _asset_is_eth(event_asset: 'Asset', pool_token: 'Asset') -> bool:
+    """Compare two assets evaluating as equal WETH and ETH"""
```
Comment (LefterisJP) [inline rotkehlchen/chain/ethereum/modules/uniswap/v3/decoder.py:93]: From the current name and description I don't understand what this did. Also isn't pool token always an EVM token?

```suggestion
def _compare_with_maybe_eth_token(asset: 'Asset', pool_token: 'EVMToken') -> bool:
    """Compares an asset with a pool token, considering ETH equal to WETH"""
```

File: rotkehlchen/chain/ethereum/modules/uniswap/v3/decoder.py [hunk 3]
```diff
@@ -425,7 +432,7 @@ def _maybe_decode_v3_deposit_or_withdrawal(
         for event in context.decoded_events:
             # search for the event of the first token
             if (
-                event.asset == resolved_assets_and_amounts[0].asset and
+                _asset_is_eth(event.asset, resolved_assets_and_amounts[0].asset) and
```
Comment (LefterisJP) [inline rotkehlchen/chain/ethereum/modules/uniswap/v3/decoder.py:435]: here use kwargs too

File: rotkehlchen/chain/ethereum/modules/uniswap/v3/decoder.py [hunk 4]
```diff
@@ -434,16 +441,17 @@ def _maybe_decode_v3_deposit_or_withdrawal(
                 event.event_type = to_event_type[0]
                 event.event_subtype = to_event_type[1]
                 event.counterparty = CPT_UNISWAP_V3
+                event_asset = event.asset.resolve_to_crypto_asset()
                 event.notes = notes.format(
                     amount=event.balance.amount,
-                    asset=resolved_assets_and_amounts[0].asset.symbol,
+                    asset=event_asset.symbol,
```
Comment (LefterisJP) [inline rotkehlchen/chain/ethereum/modules/uniswap/v3/decoder.py:447]: Why this change? Isn't `event_asset` and `resolved_assets_and_amounts[0].asset` the same thanks to the if check? This way you don't have to do the resolution you added up there.

File: rotkehlchen/chain/ethereum/modules/uniswap/v3/decoder.py [hunk 5]
```diff
@@ -452,9 +460,10 @@ def _maybe_decode_v3_deposit_or_withdrawal(
                 event.event_type = to_event_type[0]
                 event.event_subtype = to_event_type[1]
                 event.counterparty = CPT_UNISWAP_V3
+                event_asset = event.asset.resolve_to_crypto_asset()
                 event.notes = notes.format(
                     amount=event.balance.amount,
-                    asset=resolved_assets_and_amounts[1].asset.symbol,
+                    asset=event_asset.symbol,
```
Comment (LefterisJP) [inline rotkehlchen/chain/ethereum/modules/uniswap/v3/decoder.py:466]: same question here

File: rotkehlchen/chain/ethereum/modules/uniswap/v3/decoder.py (hunk 4 above)
Comment (yabirgb) [reply rotkehlchen/chain/ethereum/modules/uniswap/v3/decoder.py:447]: No because the asset could be ETH and then it would differ from the WETH that is resolved before. In `resolved_assets_and_amounts` we only have EVm tokens and in the events we can have evm tokens or ETH  I can add a if check and do the resolution only if it is eth the asset

Comment (LefterisJP) [reply rotkehlchen/chain/ethereum/modules/uniswap/v3/decoder.py:447]: Since this is already seen in the utility function you made can you modify it to return this information too? We know ETH's symbol so no need to resolve in that case.

**Review by LefterisJP — APPROVED** [review]
LGTM. Thank you Yabir
